### PR TITLE
Remove console.log() and print() requirement for challenges

### PR DESCRIPTION
**Pull request by benalleng**
Allows challenges to be completed without console logging, also improves the scripting-2 validation

(scripting-2 validation will take some more thought)

### CODE REVIEW

File: content/resources/chapter-4/public-key.tsx [hunk 1]
```diff
@@ -25,21 +24,22 @@ const encodedPrivateKey = BigInt(0x\${privateKey})
 const generatorPoint = G.mul(encodedPrivateKey)
 // Remember you need to log the answer for it to be validated
 console.log(generatorPoint)`,
```
Comment (daviroo) [inline content/resources/chapter-4/public-key.tsx:26]:
```suggestion
return generatorPoint`,
```

File: content/resources/chapter-4/public-key.tsx [hunk 2]
```diff
@@ -60,20 +60,18 @@ encoded_private_key = int(0x\${private_key})
 const generatorPoint = G.mul(encoded_private_key)
 # Remember you need to log the answer for it to be validated
 print(generator_point)`,
```
Comment (daviroo) [inline content/resources/chapter-4/public-key.tsx:62]:
```suggestion
return generator_point`,
```

File: content/resources/chapter-4/public-key.tsx (hunk 1 above)
Comment (daviroo) [reply content/resources/chapter-4/public-key.tsx:26]: There's also an error in line 22 (not made by you):

```const encodedPrivateKey = BigInt(0x\${privateKey})```

Should be:

```const encodedPrivateKey = BigInt(`0x\${privateKey}`)```

File: content/resources/chapter-4/public-key.tsx (hunk 2 above)
Comment (daviroo) [reply content/resources/chapter-4/public-key.tsx:62]: @benalleng the python solution for this challenge doesn't appear to be valid - again not your doing but we should probably fix it.

File: content/lessons/chapter-4/address-2.tsx [hunk 3]
```diff
@@ -29,7 +29,9 @@ export default function Address2({ lang }) {
   }, [])
 
   const javascript = {
-    program: `console.log("KILL")`,
+    program: `
```
Comment (daviroo) [inline content/lessons/chapter-4/address-2.tsx:32]: We need to update the solution in `address.tsx` in resources to reflect this change.

File: content/lessons/chapter-4/address-2.tsx [hunk 4]
```diff
@@ -66,14 +71,16 @@ ${prevData?.data && 'compressedPublicKey = ' + "'" + dataObject + "'"}
...
-        range: [9, 1, 9, 1],
+        range: [9, 1, 11, 1],
         allowMultiline: true,
       },
     ],
   }
 
   const python = {
-    program: `print("KILL")`,
+    program: `
```
Comment (daviroo) [inline content/lessons/chapter-4/address-2.tsx:81]: We need to update the solution in `address.tsx` in resources to reflect this change.

File: content/lessons/chapter-4/address-3.tsx [hunk 5]
```diff
@@ -30,7 +30,9 @@ export default function Address3({ lang }) {
   }, [])
 
   const javascript = {
-    program: `console.log("KILL")`,
+    program: `
```
Comment (daviroo) [inline content/lessons/chapter-4/address-3.tsx:33]: We need to update the solution in `address.tsx` in resources to reflect this change.

File: content/lessons/chapter-4/address-3.tsx [hunk 6]
```diff
@@ -71,14 +76,16 @@ ${
...
-        range: [166, 1, 166, 1],
+        range: [166, 1, 168, 1],
         allowMultiline: true,
       },
     ],
   }
 
   const python = {
-    program: `print("KILL")`,
+    program: `
```
Comment (daviroo) [inline content/lessons/chapter-4/address-3.tsx:86]: We need to update the solution in `address.tsx` in resources to reflect this change.

**Review by daviroo — CHANGES_REQUESTED** [review]

File: content/resources/chapter-4/public-key.tsx (hunk 2 above)
Comment (benalleng) [reply content/resources/chapter-4/public-key.tsx]: had an incorrect case on a param

**Review by daviroo — APPROVED** [review]

**Pull request merged**